crates/oxc_linter/src/rules/eslint/no_lone_blocks.rs

```diff
@@ -0,0 +1,375 @@
+use oxc_ast::ast::{Declaration, VariableDeclarationKind};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_lone_blocks_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Block is unnecessary.").with_label(span)
+}
+
+fn no_nested_lone_blocks_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Nested block is redundant.").with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoLoneBlocks;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallows unnecessary standalone block statements.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Standalone blocks can be confusing as they do not provide any meaningful purpose when used unnecessarily.
+    /// They may introduce extra nesting, reducing code readability, and can mislead readers about scope or intent.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// {
+    ///   var x = 1;
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (condition) {
+    ///   var x = 1;
+    /// }
+    ///
+    /// {
+    ///   let x = 1; // Used to create a valid block scope.
+    /// }
+    /// ```
+    NoLoneBlocks,
+    style,
+);
```

Comment:
I suggest adding a pending here, since this could be auto-fixed to some degree in the future:
```suggestion
    NoLoneBlocks,
    style,
    pending
);
```